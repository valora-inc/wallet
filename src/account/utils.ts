import { parsePhoneNumber } from '@celo/phone-utils'

const ADDRESS_LENGTH = 42 // TODO(ACT-1173): see if this can be replaced with a viem helper

export const isAddressFormat = (content: string): boolean => {
  return content.startsWith('0x') && content.length === ADDRESS_LENGTH
}

export function getPhoneNumberDetails(
  phoneNumber: string,
  countryCallingCode: string,
  countryCodeAlpha2: string
) {
  const phoneDetails = parsePhoneNumber(phoneNumber, countryCallingCode)

  if (phoneDetails) {
    return {
      // Show international display number to avoid confusion
      internationalPhoneNumber: phoneDetails.displayNumberInternational,
      e164Number: phoneDetails.e164Number,
      isValidNumber: true,
      countryCodeAlpha2: phoneDetails.regionCode!,
    }
  } else {
    return {
      internationalPhoneNumber: phoneNumber,
      e164Number: '',
      isValidNumber: false,
      countryCodeAlpha2,
    }
  }
}
