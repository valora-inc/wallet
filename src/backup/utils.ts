import {
  CELO_DERIVATION_PATH_BASE,
  generateKeys,
  MnemonicLanguages,
} from '@celo/cryptographic-utils'
import CryptoJS from 'crypto-js'
import { useAsync } from 'react-async-hook'
import * as bip39 from 'react-native-bip39'
import { showError } from 'src/alert/actions'
import { OnboardingEvents } from 'src/analytics/Events'
import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
import { ErrorMessages } from 'src/app/ErrorMessages'
import { getPassword } from 'src/pincode/authentication'
import { useDispatch, useSelector } from 'src/redux/hooks'
import { removeStoredItem, retrieveStoredItem, storeItem } from 'src/storage/keychain'
import Logger from 'src/utils/Logger'
import { ETHEREUM_DERIVATION_PATH } from 'src/web3/consts'
import { currentAccountSelector } from 'src/web3/selectors'

const TAG = 'Backup/utils'

const MNEMONIC_STORAGE_KEY = 'mnemonic'

export async function generateKeysFromMnemonic(mnemonic: string) {
  const wordCount = countMnemonicWords(mnemonic)
  const derivationPath = wordCount === 24 ? CELO_DERIVATION_PATH_BASE : ETHEREUM_DERIVATION_PATH
  return generateKeys(mnemonic, undefined, undefined, undefined, bip39, derivationPath)
}

export function getMnemonicLanguage(language: string | null) {
  switch (language?.slice(0, 2)) {
    case 'es': {
      return MnemonicLanguages.spanish
    }
    case 'pt': {
      return MnemonicLanguages.portuguese
    }
    default: {
      return MnemonicLanguages.english
    }
  }
}

export async function storeMnemonic(mnemonic: string, account: string | null, password?: string) {
  if (!account) {
    throw new Error('Account not yet initialized')
  }
  const passwordToUse = password ?? (await getPassword(account))
  const encryptedMnemonic = await encryptMnemonic(mnemonic, passwordToUse)
  return storeItem({ key: MNEMONIC_STORAGE_KEY, value: encryptedMnemonic })
}

export async function clearStoredMnemonic() {
  await removeStoredItem(MNEMONIC_STORAGE_KEY)
}

export async function getStoredMnemonic(
  account: string | null,
  password?: string
): Promise<string | null> {
  try {
    if (!account) {
      throw new Error('Account not yet initialized')
    }

    Logger.debug(TAG, 'Checking keystore for mnemonic')
    const encryptedMnemonic = await retrieveStoredItem(MNEMONIC_STORAGE_KEY)
    if (!encryptedMnemonic) {
      throw new Error('No mnemonic found in storage')
    }

    const passwordToUse = password ?? (await getPassword(account))
    return decryptMnemonic(encryptedMnemonic, passwordToUse)
  } catch (error) {
    Logger.error(TAG, 'Failed to retrieve mnemonic', error)
    return null
  }
}

export function onGetMnemonicFail(viewError: (error: ErrorMessages) => void, context?: string) {
  viewError(ErrorMessages.FAILED_FETCH_MNEMONIC)
  ValoraAnalytics.track(OnboardingEvents.backup_error, {
    error: 'Failed to retrieve Recovery Phrase',
    context,
  })
}

export function useAccountKey(): string | null {
  const dispatch = useDispatch()
  const account = useSelector(currentAccountSelector)
  const asyncAccountKey = useAsync(getStoredMnemonic, [account])

  if (!asyncAccountKey || asyncAccountKey.error) {
    onGetMnemonicFail((error) => dispatch(showError(error)), 'useAccountKey')
  }

  return asyncAccountKey.result || null
}

export function countMnemonicWords(phrase: string): number {
  return [...phrase.trim().split(/\s+/)].length
}

// Because of a RN bug, we can't fully clean the text as the user types
// https://github.com/facebook/react-native/issues/11068
export function formatBackupPhraseOnEdit(phrase: string) {
  return phrase.replace(/\s+/gm, ' ')
}

export function isValidBackupPhrase(phrase: string) {
  const allowedPhraseLengths = [12, 24]
  const phraseLength = countMnemonicWords(formatBackupPhraseOnEdit(phrase))
  return !!phrase && allowedPhraseLengths.includes(phraseLength)
}

export async function encryptMnemonic(phrase: string, password: string) {
  return CryptoJS.AES.encrypt(phrase, password).toString()
}

export async function decryptMnemonic(encryptedMnemonic: string, password: string) {
  const bytes = CryptoJS.AES.decrypt(encryptedMnemonic, password)
  return bytes.toString(CryptoJS.enc.Utf8)
}
