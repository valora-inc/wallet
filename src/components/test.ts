function groupNumber(value: string) {
  return value.replace(/\B(?=(\d{3})+(?!\d))(?<!\.\d*)/g, 'group')
}

function actualTest(value: string, decimalSeparator: string, groupingSeparator: string) {
  return groupNumber(value).replaceAll('.', decimalSeparator).replaceAll('group', groupingSeparator)
}

console.log(actualTest('1.234.567,9999', ',', '.')) // 1,234,567.9999
