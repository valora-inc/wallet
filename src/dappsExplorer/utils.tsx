import { DappV2WithCategoryNames } from 'src/dapps/types'

/***
 * Calculates a score based on the search term and dapp
 * - +2 if the name matches the search term in a case sensitive way
 * - +1 if the name matches the search term in a case insensitive way
 * - +1 if the description matches the search term in a case insensitive way
 * - +1 if the category matches the search term in a case insensitive way
 * @param dapp: dapp with categoryNames
 * @param searchTerm: the user input search string
 * @returns {number} the score of the dapp 0 - 5
 */
export const calculateSearchScore = (dapp: DappV2WithCategoryNames, searchTerm: string) => {
  if (searchTerm === '') return 0
  const nameMatchingCaseScore = dapp.name.includes(searchTerm) ? 2 : 0
  const nameScore = dapp.name.toLowerCase().includes(searchTerm.toLowerCase()) ? 1 : 0
  const descriptionScore = dapp.description.toLowerCase().includes(searchTerm.toLowerCase()) ? 1 : 0
  const categoryScore = dapp.categoryNames.some((category) =>
    category.toLowerCase().includes(searchTerm.toLowerCase())
  )
    ? 1
    : 0
  return nameMatchingCaseScore + nameScore + descriptionScore + categoryScore
}
