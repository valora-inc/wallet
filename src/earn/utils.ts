import BigNumber from 'bignumber.js'
import { getFeatureGate } from 'src/statsig'
import { StatsigFeatureGates } from 'src/statsig/types'
import { SwapTransaction } from 'src/swap/types'
import { Network, NetworkId } from 'src/transactions/types'
import { INTERNAL_RPC_SUPPORTED_NETWORKS } from 'src/viem'
import { networkIdToNetwork } from 'src/web3/networkConfig'

/**
 * Helper function to check if gas is subsidized for the given networkId
 * @param networkId - networkId to check
 * @returns boolean
 */
export function isGasSubsidizedForNetwork(networkId: NetworkId) {
  return (
    (INTERNAL_RPC_SUPPORTED_NETWORKS as readonly Network[]).includes(
      networkIdToNetwork[networkId]
    ) && getFeatureGate(StatsigFeatureGates.SUBSIDIZE_STABLECOIN_EARN_GAS_FEES)
  )
}

/**
 * Gets the swap to amount in decimals given a swap transaction and the swap
 * from amount in decimals. This could also be taken from the buy amount, but
 * that is in wei and we would need the token decimals. This follows what is
 * done in the SwapScreen
 * @param object.swapTransaction - swap transaction object
 * @param object.fromAmount - swap from amount in decimals (usually from the user input)
 * @returns swap to amount in decimals
 */
export function getSwapToAmountInDecimals({
  swapTransaction,
  fromAmount,
}: {
  swapTransaction: SwapTransaction
  fromAmount: BigNumber
}) {
  return fromAmount.multipliedBy(swapTransaction.price)
}
