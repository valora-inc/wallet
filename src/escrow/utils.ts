import { EscrowedPayment } from 'src/escrow/actions'
import { identifierToE164NumberSelector } from 'src/identity/selectors'
import { ContactRecipient } from 'src/recipients/recipient'
import { useSelector } from 'src/redux/hooks'

export const useEscrowPaymentRecipient = (payment: EscrowedPayment): ContactRecipient => {
  const { recipientPhone, recipientIdentifier } = payment

  const identifierToE164Number = useSelector(identifierToE164NumberSelector)
  const recipientCache = useSelector((state) => state.recipients.phoneRecipientCache)
  const phoneNumber = identifierToE164Number[recipientIdentifier] ?? recipientPhone
  const recipient = recipientCache[phoneNumber] ?? {
    name: phoneNumber,
    e164PhoneNumber: phoneNumber,
    contactId: '',
  }

  return recipient
}
