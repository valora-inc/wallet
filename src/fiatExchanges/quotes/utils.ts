import { SettlementEstimation, SettlementTime } from 'src/fiatExchanges/quotes/constants'
import Logger from 'src/utils/Logger'

const TAG = 'fiatExchanges/quotes/utils'

export const getSettlementTimeString = (
  settlementEstimation: SettlementEstimation,
  settlementTimeStrings: Record<SettlementTime, string>
) => {
  switch (settlementEstimation.settlementTime) {
    case SettlementTime.LESS_THAN_ONE_HOUR:
      return { timeString: settlementTimeStrings[SettlementTime.LESS_THAN_ONE_HOUR] }
    case SettlementTime.LESS_THAN_X_HOURS:
    case SettlementTime.LESS_THAN_X_DAYS:
      return {
        timeString: settlementTimeStrings[settlementEstimation.settlementTime],
        upperBound: settlementEstimation.upperBound,
      }
    case SettlementTime.X_TO_Y_HOURS:
    case SettlementTime.X_TO_Y_DAYS:
      return {
        timeString: settlementTimeStrings[settlementEstimation.settlementTime],
        lowerBound: settlementEstimation.lowerBound,
        upperBound: settlementEstimation.upperBound,
      }
    default:
      // Should not be reachable
      Logger.warn(TAG, 'Unexpected settlement time value in settlement estimation')
      return {}
  }
}
