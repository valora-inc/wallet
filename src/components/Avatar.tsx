import * as React from 'react'
import { WithTranslation } from 'react-i18next'
import { StyleSheet, Text, TextStyle, View } from 'react-native'
import { useSelector } from 'react-redux'
import { defaultCountryCodeSelector } from 'src/account/selectors'
import ContactCircle from 'src/components/ContactCircle'
import PhoneNumberWithFlag from 'src/components/PhoneNumberWithFlag'
import { formatShortenedAddress } from 'src/components/ShortenedAddress'
import { withTranslation } from 'src/i18n'
import { getDisplayName, Recipient } from 'src/recipients/recipient'
import fontStyles from 'src/styles/fonts'

const DEFAULT_ICON_SIZE = 40

interface OwnProps {
  recipient: Recipient
  e164Number?: string
  iconSize?: number
  displayNameStyle?: TextStyle
}

type Props = OwnProps & WithTranslation

export function Avatar(props: Props) {
  const defaultCountryCode = useSelector(defaultCountryCodeSelector) ?? undefined
  const { recipient, e164Number, iconSize = DEFAULT_ICON_SIZE, displayNameStyle, t } = props

  const name = getDisplayName(recipient, t)
  const address = recipient.address
  const e164NumberToShow = recipient.e164PhoneNumber || e164Number

  return (
    <View style={styles.container}>
      <ContactCircle recipient={recipient} size={iconSize} />
      <Text
        style={[displayNameStyle || fontStyles.small500, styles.contactName]}
        numberOfLines={1}
        ellipsizeMode="tail"
      >
        {name}
      </Text>
      {e164NumberToShow && (
        <PhoneNumberWithFlag
          e164PhoneNumber={e164NumberToShow}
          defaultCountryCode={defaultCountryCode}
        />
      )}
      {!e164NumberToShow && address && (
        <Text style={[fontStyles.small, styles.contactName]} numberOfLines={1} ellipsizeMode="tail">
          {formatShortenedAddress(address)}
        </Text>
      )}
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    marginBottom: 10,
  },
  contactName: {
    paddingTop: 6,
    marginHorizontal: 20,
    textAlign: 'center',
  },
})

export default withTranslation<Props>()(Avatar)
