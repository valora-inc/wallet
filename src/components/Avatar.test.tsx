import { render } from '@testing-library/react-native'
import * as React from 'react'
import { Provider } from 'react-redux'
import Avatar from 'src/components/Avatar'
import { createMockStore } from 'test/utils'

const mockName = 'mockName'
const mockNumber = '+14155556666'
const mockAccount = '0x0000000000000000000000000000000000007E57'

const store = createMockStore({
  account: {
    defaultCountryCode: '+1',
  },
})

describe(Avatar, () => {
  it('renders correctly with number and name', () => {
    const tree = render(
      <Provider store={store}>
        <Avatar recipient={{ name: mockName, e164PhoneNumber: mockNumber }} iconSize={40} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with just number', () => {
    const tree = render(
      <Provider store={store}>
        <Avatar recipient={{ e164PhoneNumber: mockNumber }} iconSize={40} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with address and name but no number', () => {
    const tree = render(
      <Provider store={store}>
        <Avatar recipient={{ name: mockName, address: mockAccount }} iconSize={40} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with address but no name nor number', () => {
    const tree = render(
      <Provider store={store}>
        <Avatar recipient={{ address: mockAccount }} iconSize={40} />
      </Provider>
    )
    expect(tree).toMatchSnapshot()
  })
})
