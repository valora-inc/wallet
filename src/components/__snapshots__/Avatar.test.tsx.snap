// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly with address and name but no number 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "backgroundColor": "hsl(0, 53%, 93%)",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          [
            {
              "color": "#FFFFFF",
              "fontFamily": "Inter-Medium",
              "fontSize": 16,
            },
            {
              "color": "hsl(0, 67%, 24%)",
              "fontSize": 20,
            },
          ]
        }
      >
        M
      </Text>
    </View>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "color": "#2E3338",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "marginHorizontal": 20,
          "paddingTop": 6,
          "textAlign": "center",
        },
      ]
    }
  >
    mockName
  </Text>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "color": "#2E3338",
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "marginHorizontal": 20,
          "paddingTop": 6,
          "textAlign": "center",
        },
      ]
    }
  >
    0x0000...7E57
  </Text>
</View>
`;

exports[`Avatar renders correctly with address but no name nor number 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "backgroundColor": "hsl(0, 53%, 93%)",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          },
        ]
      }
    >
      <svg
        fill="none"
        height="40"
        viewBox="0 0 40 40"
        width="40"
      >
        <circle
          cx="20"
          cy="20"
          fill="hsl(0, 53%, 93%)"
          r="20"
        />
        <path
          d="M20 12a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 2a2 2 0 1 0 0 4 2 2 0 0 0 0-4Zm0 7c2.67 0 8 1.33 8 4v3H12v-3c0-2.67 5.33-4 8-4Zm0 1.9c-2.97 0-6.1 1.46-6.1 2.1v1.1h12.2V25c0-.64-3.13-2.1-6.1-2.1Z"
          fill="hsl(0, 67%, 24%)"
        />
      </svg>
    </View>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "color": "#2E3338",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "marginHorizontal": 20,
          "paddingTop": 6,
          "textAlign": "center",
        },
      ]
    }
  >
    feedItemAddress, {"address":"0x0000...7E57"}
  </Text>
</View>
`;

exports[`Avatar renders correctly with just number 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "backgroundColor": "hsl(0, 53%, 93%)",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          },
        ]
      }
    >
      <svg
        fill="none"
        height="40"
        viewBox="0 0 40 40"
        width="40"
      >
        <circle
          cx="20"
          cy="20"
          fill="hsl(0, 53%, 93%)"
          r="20"
        />
        <path
          d="M20 12a4 4 0 1 1 0 8 4 4 0 0 1 0-8Zm0 2a2 2 0 1 0 0 4 2 2 0 0 0 0-4Zm0 7c2.67 0 8 1.33 8 4v3H12v-3c0-2.67 5.33-4 8-4Zm0 1.9c-2.97 0-6.1 1.46-6.1 2.1v1.1h12.2V25c0-.64-3.13-2.1-6.1-2.1Z"
          fill="hsl(0, 67%, 24%)"
        />
      </svg>
    </View>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "color": "#2E3338",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "marginHorizontal": 20,
          "paddingTop": 6,
          "textAlign": "center",
        },
      ]
    }
  >
    +14155556666
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#2E3338",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginRight": 4,
          },
        ]
      }
    >
      ðŸ‡ºðŸ‡¸
    </Text>
    <Text
      style={
        {
          "color": "#2E3338",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      +1 415-555-6666
    </Text>
  </View>
</View>
`;

exports[`Avatar renders correctly with number and name 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 10,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "backgroundColor": "hsl(0, 53%, 93%)",
            "borderRadius": 20,
            "height": 40,
            "width": 40,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          [
            {
              "color": "#FFFFFF",
              "fontFamily": "Inter-Medium",
              "fontSize": 16,
            },
            {
              "color": "hsl(0, 67%, 24%)",
              "fontSize": 20,
            },
          ]
        }
      >
        M
      </Text>
    </View>
  </View>
  <Text
    ellipsizeMode="tail"
    numberOfLines={1}
    style={
      [
        {
          "color": "#2E3338",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "lineHeight": 18,
        },
        {
          "marginHorizontal": 20,
          "paddingTop": 6,
          "textAlign": "center",
        },
      ]
    }
  >
    mockName
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#2E3338",
            "fontFamily": "Inter-Regular",
            "fontSize": 14,
            "lineHeight": 18,
          },
          {
            "alignItems": "center",
            "justifyContent": "center",
            "marginRight": 4,
          },
        ]
      }
    >
      ðŸ‡ºðŸ‡¸
    </Text>
    <Text
      style={
        {
          "color": "#2E3338",
          "fontFamily": "Inter-Medium",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      +1 415-555-6666
    </Text>
  </View>
</View>
`;
